src/settings.c

```diff
@@ -781,6 +1044,9 @@ static setting_data_t *setting_create_setting(settings_t *ctx,
     .next = NULL,
   };
 
+  strncpy(setting_data->section, section, strlen(section) + 1);
```

Comment:
Here we use `setting_data->section` and `setting_data->name` before they are checked for NULL at the line 1050 below.